mariamihaelafainarea/assignment2/LowestAncestor

```diff
@@ -1,100 +1,80 @@
-                return root;
-            }
-            //if here are both x and y propagate root.
-            if(left != null && right != null){
-                return root;
-            }
-            //propagate what is not null
-            if (left != null) {
-                return left;
-            }
-            else if (right != null) {
-               return right;
-           }
+public class LowestAncestor<T> {
+     
+    Node first;
+    //these parameters help me find
+    //if the two values searched were found
+        
+   
+    private Node LowestAncestorHelper(T x, Node root,T y,AtomicBoolean params1,AtomicBoolean params2){
+        //it is a recursive searching
+        //of the x,y.
+        if (root == null) {
             return null;
         }
-        public Node isARealAncestor(T x,T y) {
-            //this functions and param1,param2 verifies if "findLowstAncestor"
-            //is correct because it can happen
-            //to have an x or an y that do not exists
-            //in the tree.     
-            AtomicBoolean params1 = new AtomicBoolean(false);
-            AtomicBoolean params2 = new AtomicBoolean(false);
-            Node result = findLowestAncestor(x, first, y,params1,params2);
-            if (result == null || !(params1.get() == true && params2.get() == true)) {
-                System.out.println("I haven't found what you've been looking!");
-                return null;
-            } else if (params1.get() == true && params2.get() == true){
-            System.out.println("Lowest ancestor of " 
-                    + x + " and " + y + " is: " + result.key);
-                    return result;
-            }
+        if (root.getKey().equals(x)) {
+            params1.set(true);
+        }
+        if (root.getKey().equals(y)) {
+            params2.set(true);
+        }
+        Node left = LowestAncestorHelper(x, root.getLeft(), y,params1,params2); 
+        Node right = LowestAncestorHelper(x, root.getRight(), y,params1,params2); 
+        if(root.getKey().equals(x) || root.getKey().equals(y)){
+            return root;
+        }
+        //if here are both x and y propagate root.
+        if(left != null && right != null){
+            return root;
+        }
+        //propagate what is not null
+        if (left != null) {
+            return left;
+        }
+        else if (right != null) {
+           return right;
+       }
+        return null;
+    }
+    public Node lowestCommonAncestor(T x,T y) {
+        //this functions and param1,param2 verifies if "findLowstAncestor"
+        //is correct because it can happen
+        //to have an x or an y that do not exists
+        //in the tree.     
+        AtomicBoolean params1 = new AtomicBoolean(false);
+        AtomicBoolean params2 = new AtomicBoolean(false);
+        Node result = LowestAncestorHelper(x, first, y,params1,params2);
+        if (result == null || !(params1.get() == true && params2.get() == true)) {
```

Comment:
maybe just replace it with if (result == null || !params1.get() || !params2.get())

Just checking for 2 ORs is easier to read plus you don't need to compare boolean to true or false to get a boolean ;)